<a href="https://colab.research.google.com/github/matheus97eng/imersao-dados-desafio-final/blob/main/Notebooks/Projeto_Final_Matheus_Maia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Projeto Imersão Dados Alura - 3ª ed.

### Participante: 

Matheus Maia Ávila

Estudante de Engenharia Física (UFG)

---



# Sobre a imersão

A terceira edição da Imersão Dados da Alura aconteceu nos dias 3 a 7 de Maio de 2021...

# Sobre o projeto



## Motivações

O avanço da ciência e o crescimento da biotecnologia são essenciais no estudo farmacológico. A própria situação crítica de saúde pública que recentemente passamos por conta do coronavírus mostra o quão importante é o avanço de pesquisas científicas que não só nos ajudam a entender o problema epidemiológico que passamos, mas também nos dá ferramentas para combatê-lo, inclusive com a criação de vacinas.

De maneira geral, a prospecção de novos fármacos passa pelo estudo científico, fazendo-se testes da ação das drogas em organismos, sejam essas drogas novos compostos ou medicamentos já conhecidos, mas que podem fornecer efeitos interessantes ainda não descobertos. Esses estudos são em sua maioria despendiosos e, principalmente por conta do avanço tecnológico, produzem uma grande quantidade de dados. Esses dados precisam ser analisados, e então entra o papel da ciência de dados.

Em uma dessas situações, o Connectivity Map, projeto que reúne as instituições MIT, Harvard e LINCS, lançou um desafio no kaggle sobre como o machine learning pode ajudar na predição do número de mecanismos de ativação de uma droga. Esse desafio que inspirou a imersão e baseia este projeto.

## Objetivos

Tendo em vista a contextualização acima, este projeto 

# Conceitos importantes

**viabilidade celular**: uma maneira de avaliar a atividade das células nas culturas. Há várias formas de se medir a viabilidade celular, uma delas é a medição de ATP (adenosina trifosfato) intracelular

# Código

In [14]:
# importação de bibliotecas

import pandas as pd

## Importando dados

Os dados foram tirados do repositório no github da Alura, através de um endereço url. O primeiro dataset contém as informações de como o experimento foi feito e as expressões gênicas obtidas.

In [15]:
url_dados = 'https://github.com/alura-cursos/imersaodados3/blob/main/dados/dados_experimentos.zip?raw=true'

df = pd.read_csv(url_dados, compression = 'zip')
print(df.shape)
df.head()

(23814, 877)


id tratamento  tempo dose  ...    c-96    c-97    c-98    c-99
0  id_000644bb2  com_droga     24   D1  ... -0.3981  0.2139  0.3801  0.4176
1  id_000779bfc  com_droga     72   D1  ...  0.1522  0.1241  0.6077  0.7371
2  id_000a6266a  com_droga     48   D1  ... -0.6417 -0.2187 -1.4080  0.6931
3  id_0015fd391  com_droga     48   D1  ... -1.6210 -0.8784 -0.3876 -0.8154
4  id_001626bd3  com_droga     72   D2  ...  0.1094  0.2885 -0.3786  0.7125

[5 rows x 877 columns]

Como podemos ver, o dataset possui 23814 linhas (experimentos) e 877 colunas. Vamos compreender melhor o que descrevem essas colunas:

* **id**: identificação única do experimento;
* **tratamento**: o tipo de tratamento que o experimento levou, se com droga ou sem (teste de controle);
* **tempo**: tempo de duração do experimento, verificando o tempo de ação nas células, se de 24h, 48h ou 72h;
* **dose**: se a dose foi do tipo D1 ou D2. D1 significa dose alta e D2 significa dose baixa
* **g-0 a g-771**: colunas que informam a expressão gênica encontrada no experimento, dependendo do gene.
* **c-0 a c-99**: colunas que remetem à viabilidade celular.

Através de alguns comandos, podemos verificar essas informações referente às colunas


In [16]:
# verificando se os id são realmente únicos
len(df['id'].unique()) == 23814  # cada linha possui um id diferente

True

In [17]:
# tipos de dados da coluna tratamento
df['tratamento'].unique()

array(['com_droga', 'com_controle'], dtype=object)

In [18]:
# tipos de dados da coluna tempo
df['tempo'].unique()

array([24, 72, 48])

In [19]:
# tipos de dados da coluna dose
df['dose'].unique()

array(['D1', 'D2'], dtype=object)

Outro dataset importante é o de resultados. Ele é importado também através do repositório da Alura.

In [20]:
df_resultados = pd.read_csv('https://github.com/alura-cursos/imersaodados3/blob/main/dados/dados_resultados.csv?raw=true')
print(df_resultados.shape)
df_resultados.head()

id  ...  wnt_inhibitor
0  id_000644bb2  ...              0
1  id_000779bfc  ...              0
2  id_000a6266a  ...              0
3  id_0015fd391  ...              0
4  id_001626bd3  ...              0

[5 rows x 207 columns]

Ele possui a mesma coluna id dos experimentos. As outras 206 colunas dizem respeito ao mecanismo de ação nas células. Cada uma dessas colunas representa um mecanismo de ação diferente na célula. A descrição do mecanismo contém primeiramente o alvo do mecanismo e também o tipo de ação causada. Por exemplo, na coluna 5-alpha_reductase_inhibitor o alvo é a enzima 5-alpha_reductase e ação causada é inibidora.

Podemos ver também que os valores dessas colunas são sempre 0 ou 1. Valor 1 significa que o mecanismo agiu e valor 0, que não. Vamos verificar com código.

In [35]:
# verificando se os id são realmente únicos
len(df_resultados['id'].unique()) == 23814  # cada linha possui um id diferente

True

In [57]:
#verificando os valores dos mecanismos de ativação
eh_0ou1 = True
for col in df_resultados.drop('id', axis=1).columns:
    if df_resultados[col].unique().sum() != 1:
      eh_0ou1 = False

eh_0ou1
# se df_resultados[col].unique() é sempre [0,1] ou [1,0], então a soma dessa array será sempre = 1

True

# Bibliografia

## Links 

* https://www.promega.com.br/c/blog-jovem-pesquisador/articles/teste-de-viabilidade-celular/